File: js/parser/util/PeekTokenIterator.js
const ParseException = require("./ParseException");
const PeekIterator =require("../../common/PeekIterator")
class PeekTokenIterator extends PeekIterator{
    constructor(it) {
        super(it);
    }

    nextMatch(value){
        let token = this.next()
        if (token.getValue() !==value){
            throw new ParseException.formToken(token)
        }
        return token
    }

    nextMatch1(type){
        let token = this.next()
        if (token.getType() !==type){
            throw new ParseException.formToken(token)
        }
        return token
    }

}

module.exports=PeekTokenIterator;

File: java/src/main/java/parser/utils/PeekTokenIterator.java
package parser.utils;

import common.PeekIterator;
import lexer.Token;
import lexer.TokenType;

import java.util.stream.Stream;

public class PeekTokenIterator  extends PeekIterator<Token> {
    public PeekTokenIterator(Stream<Token> stream) {
        super(stream);
    }


    public PeekTokenIterator(Stream<Token> stream, Object endToKen) {
        super(stream, (Token) endToKen);
    }

    public Token nextMatch(String value) throws ParseException {
        Token token = this.next();
        if (!token.getValue().equals(value)){
            throw new ParseException(token);
        }
        return token;
    }

    public Token nextMatch(TokenType type) throws ParseException {
        Token token =this.next();
        if (token.getType()==type){
            throw new ParseException(token);
        }
        return token;
    }
}
